pcdsdevices/lasers/qmini.py

```diff
@@ -83,6 +87,47 @@ class QminiSpectrometer(Device):
     fit_stdev = Cpt(EpicsSignalRO, ':STDEV', kind='config')
     fit_chisq = Cpt(EpicsSignalRO, ':CHISQ', kind='config')
 
+    # Save spectra functions
+    def save_data(self):
+        """
+        Save the wavelength and spectrum PVs to a text file
+        """
+        if len(self.file_dest.get().strip()) == 0:
```

Comment:
Nitpick: empty strings evaluate to False without needing to get the length.
```suggestion
        if not self.file_dest.get().strip():
```

Reply:
Huh! I didn't realize that, and it is _much_ cleaner